Power Menu Loads No Controls When Any Server is Inaccessible

<!-- READ THIS FIRST:
- Make sure you run the latest version of the Android app
- Make sure you run the latest version of Home Assistant
- Make sure to check the Companion docs for troubleshooting and configuration: https://companion.home-assistant.io/
- Make sure the bug you found is not already reported, we love to put work in bugfixes instead of closing duplicate bug reports
  DO NOT DELETE ANY TEXT from this template! All requested information is important.
-->

<!-- If you are reporting an issue for Wear OS please list both devices for the below 3 fields -->
**Home Assistant Android app version(s):** version 2024.1.5-full

**Android version(s):** 11

**Device model(s):** Moto G Stylus 2022 (XT2043-4)

**Home Assistant version:** 2024.2.5

**Last working Home Assistant release (if known):** Unknown (2023+?)

**Description of problem, include YAML if issue is related to notifications:**

I added Power Menu controls (hold power button and get access to smart home controls quickly) a long time ago, and didn't really mess with them for a long time (though I think I did sometime in 2023). I recently renamed a bunch of entities and needed to replace a few controls. I removed 2 controls (still have 5 old controls that work), but I cannot add any new controls. It tries to load for a little while, then  shows the screen below. I have tried clearing cache and restarting my phone, but no dice.

Adding Power Menu controls seems to be broken for me, and I am not sure if it is a broader issue for other users or not.

Thanks for the help.
Nick

<!--
- For Wear OS devices we will need the LogCat logs from the device.
- For Android Auto the logs can be retrieved from the connected device.
- Logs from the device can be taken from Settings > Companion App Troubleshooting > Show and Share Logs
-->
**Companion App Logs:**

```

```

**Screenshot or video of problem:**
![Screenshot_20240305-070339.png](https://github.com/home-assistant/android/assets/918497/68634338-1066-4973-bdba-b2c9019d391b)

**Additional information:**

Sorry for the delay, I had to get home to set everything back up and try to reproduce the problem.

I didn't manage to capture the log the first time around (I was not in a position to scrub it at that time), and confusingly now it seems to act how we would expect (it shows devices from whatever server(s) it can connect to). Sorry about that. Perhaps somehow I was in a funky state that I can't reproduce at the moment. However, I do still have some errors in the logs that look similar to what I saw earlier, maybe they will still be helpful. And I can try again over the next few days.

Quick notes about my setup. I have two HA instances: the primary is set up for HTTPS via a reverse proxy, and the secondary is set up for HTTP on the bare :8123 port. I use Tailscale/Headscale as a VPN from my phone to my house, and apparently mis-configured it such that the secondary server is inaccessible over the VPN (gotta figure that out another time). My gut feeling is that it was confused that it had been connected to the secondary server at one point in time, then was unable to re-connect later.

```
--------- beginning of main
03-05 18:04:44.935 16188 16188 D SensorReceiver: Received intent: android.net.wifi.WIFI_STATE_CHANGED
03-05 18:04:44.937 16188 16188 D SensorReceiver: Sensor wifi_state corresponding to received event android.net.wifi.WIFI_STATE_CHANGED is disabled, skipping sensors update
03-05 18:04:44.952 16188 16188 D SensorReceiver: Received intent: android.net.wifi.WIFI_STATE_CHANGED
03-05 18:04:44.960 16188 16188 D SensorReceiver: Sensor wifi_state corresponding to received event android.net.wifi.WIFI_STATE_CHANGED is disabled, skipping sensors update
03-05 18:04:45.193 16188 16188 D SensorReceiver: Received intent: android.net.wifi.WIFI_STATE_CHANGED
03-05 18:04:45.194 16188 16188 D SensorReceiver: Sensor wifi_state corresponding to received event android.net.wifi.WIFI_STATE_CHANGED is disabled, skipping sensors update
03-05 18:04:45.367 16188 16188 D SensorReceiver: Received intent: android.net.wifi.STATE_CHANGE
03-05 18:04:45.389 16188  6461 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: true
03-05 18:04:45.808 16188  6460 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:04:45.809 16188  6463 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:04:45.810 16188  6460 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:04:45.813 16188  6463 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:04:47.754 16188 16188 D IntegrationRepository: isAppLocked(): false. (LockEnabled: false, appActive: false, expireMillis: 1709679852966, currentMillis: 1709679887753)
03-05 18:04:47.812 16188 16188 D IntegrationRepository: isAppLocked(): false. (LockEnabled: false, appActive: false, expireMillis: 1709679852966, currentMillis: 1709679887812)
03-05 18:04:47.813 16188 16188 D IntegrationRepository: setAppActive(): true
03-05 18:04:49.772 16188  6463 D SensorReceiver: Nothing to update for server 1 (Home)
03-05 18:04:50.827 16188 16188 D IntegrationRepository: setAppActive(): false
03-05 18:04:50.828 16188 16188 D IntegrationRepository: setAppActive(): false
03-05 18:04:58.375 16188  6463 E SensorReceiver: Error while getting core config to sync sensor status
03-05 18:04:58.375 16188  6463 E SensorReceiver: io.homeassistant.companion.android.common.data.integration.IntegrationException: java.net.SocketTimeoutException: failed to connect to home-assistant.iot.domain/10.202.0.13 (port 8123) from /100.64.0.1 (port 48194) after 10000ms
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at io.homeassistant.companion.android.common.data.integration.impl.IntegrationRepositoryImpl.getConfig(IntegrationRepositoryImpl.kt:533)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at io.homeassistant.companion.android.common.data.integration.impl.IntegrationRepositoryImpl$getConfig$1.invokeSuspend(Unknown Source:14)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:103)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
03-05 18:04:58.375 16188  6463 E SensorReceiver: Caused by: java.net.SocketTimeoutException: failed to connect to home-assistant.iot.domain/10.202.0.13 (port 8123) from /100.64.0.1 (port 48194) after 10000ms
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at libcore.io.IoBridge.connectErrno(IoBridge.java:190)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at libcore.io.IoBridge.connect(IoBridge.java:134)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at java.net.PlainSocketImpl.socketConnect(PlainSocketImpl.java:142)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:390)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:230)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:212)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:436)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at java.net.Socket.connect(Socket.java:621)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.platform.Platform.connectSocket(Platform.kt:128)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:295)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
03-05 18:04:58.375 16188  6463 E SensorReceiver: 	at java.lang.Thread.run(Thread.java:923)
03-05 18:04:58.671 16188  6463 D SensorReceiver: Nothing to update for server 3 (IOT)
03-05 18:04:58.672 16188  6463 I SensorReceiver: Sensor updates and sync completed
03-05 18:04:59.200 16188  6463 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:04:59.211 16188  6463 I chatty  : uid=10215(io.homeassistant.companion.android) DefaultDispatch identical 2 lines
03-05 18:04:59.215 16188  6463 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:04:59.217 16188  6460 D WebSocketRepository: Sending message 886: {type=ping, id=886}
03-05 18:04:59.218 16188  6463 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:04:59.219 16188  6460 D WebSocketRepository: Message number 886 sent
03-05 18:04:59.290 16188 30184 D WebSocketRepository: Websocket: onMessage (text)
03-05 18:04:59.292 16188 30184 D WebSocketRepository: Message number 886 received
03-05 18:05:01.108 16188 16188 D IntegrationRepository: isAppLocked(): false. (LockEnabled: false, appActive: false, expireMillis: 1709679891826, currentMillis: 1709679901108)
03-05 18:05:01.191 16188 16188 D IntegrationRepository: isAppLocked(): false. (LockEnabled: false, appActive: false, expireMillis: 1709679891826, currentMillis: 1709679901191)
03-05 18:05:01.191 16188 16188 D IntegrationRepository: setAppActive(): true
03-05 18:05:05.696 16188 16188 D HaConProService: publisherFor [1.climate.iot_house_thermostat, 1.climate.master_bathroom, 1.fan.master_bedroom_fan, 1.light.zigbee_group_master_lamp_bulbs, light.family_room_lights, 1.light.piggie_light, 1.light.outside_lights, 1.light.plug_c]
03-05 18:05:05.736 16188  3121 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:05:05.736 16188  3121 D HaConProService: request 100000
03-05 18:05:05.738 16188 31029 D WebSocketRepository: Sending message 887: {type=config/entity_registry/list, id=887}
03-05 18:05:05.739 16188 31029 D WebSocketRepository: Message number 887 sent
03-05 18:05:05.739 16188  6453 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:05:05.739 16188  6453 D ServerConnectionInfo: Using external URL
03-05 18:05:05.739 16188  6460 D WebSocketRepository: Sending message 888: {type=config/device_registry/list, id=888}
03-05 18:05:05.740 16188  6460 D WebSocketRepository: Message number 888 sent
03-05 18:05:05.740 16188  6454 D WebSocketRepository: Sending message 889: {type=config/area_registry/list, id=889}
03-05 18:05:05.740 16188  6454 D WebSocketRepository: Message number 889 sent
03-05 18:05:07.365 16188 30184 D WebSocketRepository: Websocket: onMessage (text)
03-05 18:05:07.467 16188 30184 D WebSocketRepository: Message number 887 received
03-05 18:05:07.542 16188 16202 I mpanion.androi: Background young concurrent copying GC freed 192655(8423KB) AllocSpace objects, 34(4160KB) LOS objects, 23% free, 24MB/31MB, paused 398us total 122.330ms
03-05 18:05:07.618 16188 30184 D WebSocketRepository: Websocket: onMessage (text)
03-05 18:05:07.633 16188 30184 D WebSocketRepository: Message number 888 received
03-05 18:05:07.634 16188 30184 D WebSocketRepository: Message number 889 received
03-05 18:05:07.644 16188  6456 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:05:07.645 16188 31029 D WebSocketRepository: Sending message 890: {type=subscribe_events, event_type=area_registry_updated, id=890}
03-05 18:05:07.646 16188 31029 D WebSocketRepository: Message number 890 sent
03-05 18:05:07.723 16188 30184 D WebSocketRepository: Websocket: onMessage (text)
03-05 18:05:07.723 16188 30184 D WebSocketRepository: Message number 890 received
03-05 18:05:07.726 16188 31029 D WebSocketRepository: Sending message 891: {type=subscribe_events, event_type=entity_registry_updated, id=891}
03-05 18:05:07.726 16188 31029 D WebSocketRepository: Message number 891 sent
03-05 18:05:07.790 16188 30184 D WebSocketRepository: Websocket: onMessage (text)
03-05 18:05:07.791 16188 30184 D WebSocketRepository: Message number 891 received
03-05 18:05:07.795 16188  6456 D WebSocketRepository: Sending message 892: {type=subscribe_events, event_type=device_registry_updated, id=892}
03-05 18:05:07.795 16188  6456 D WebSocketRepository: Message number 892 sent
03-05 18:05:07.847 16188 30184 D WebSocketRepository: Websocket: onMessage (text)
03-05 18:05:07.848 16188 30184 D WebSocketRepository: Message number 892 received
03-05 18:05:07.851 16188  6456 D WebSocketRepository: Sending message 893: {type=subscribe_entities, entity_ids=[climate.iot_house_thermostat, climate.master_bathroom, fan.master_bedroom_fan, light.zigbee_group_master_lamp_bulbs, light.family_room_lights, light.piggie_light, light.outside_lights, light.plug_c], id=893}
03-05 18:05:07.852 16188  6456 D WebSocketRepository: Message number 893 sent
03-05 18:05:07.916 16188 30184 D WebSocketRepository: Websocket: onMessage (text)
03-05 18:05:07.918 16188 30184 D WebSocketRepository: Message number 893 received
03-05 18:05:07.918 16188 30184 D WebSocketRepository: Message number 893 received
03-05 18:05:07.924 16188  6462 D HaConProService: Sending 8 entities to subscriber
03-05 18:05:09.669 16188 16188 D IntegrationRepository: setAppActive(): false
03-05 18:05:09.670 16188 16188 D IntegrationRepository: setAppActive(): false
03-05 18:05:09.809 16188  6463 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:05:09.811 16188  6452 D WebSocketRepository: Sending message 894: {type=config/device_registry/list, id=894}
03-05 18:05:09.811 16188  6452 D WebSocketRepository: Message number 894 sent
03-05 18:05:09.812 16188  6456 D WebSocketRepository: Sending message 895: {type=config/area_registry/list, id=895}
03-05 18:05:09.812 16188  6456 D WebSocketRepository: Message number 895 sent
03-05 18:05:09.813 16188  6463 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:05:09.813 16188  6463 D ServerConnectionInfo: Using external URL
03-05 18:05:09.814 16188  6452 D WebSocketRepository: Sending message 896: {type=get_states, id=896}
03-05 18:05:09.814 16188  6452 D WebSocketRepository: Message number 896 sent
03-05 18:05:09.817 16188  6462 D WebSocketRepository: Sending message 897: {type=config/entity_registry/list, id=897}
03-05 18:05:09.818 16188  6462 D WebSocketRepository: Message number 897 sent
03-05 18:05:09.819 16188  6463 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:05:09.819 16188  6463 D ServerConnectionInfo: Using external URL
03-05 18:05:09.902 16188  3121 D HaConProService: cancel
03-05 18:05:09.909 16188 31028 D WebSocketRepository: Unsubscribing from subscribe_entities with data {entity_ids=[climate.iot_house_thermostat, climate.master_bathroom, fan.master_bedroom_fan, light.zigbee_group_master_lamp_bulbs, light.family_room_lights, light.piggie_light, light.outside_lights, light.plug_c]}
03-05 18:05:09.909 16188 31028 D WebSocketRepository: Sending message 898: {type=unsubscribe_events, subscription=893, id=898}
03-05 18:05:09.910 16188 31028 D WebSocketRepository: Message number 898 sent
03-05 18:05:09.912 16188  9648 D WebSocketRepository: Unsubscribing from subscribe_events with data {event_type=area_registry_updated}
03-05 18:05:09.913 16188  9648 D WebSocketRepository: Sending message 899: {type=unsubscribe_events, subscription=890, id=899}
03-05 18:05:09.913 16188  9648 D WebSocketRepository: Message number 899 sent
03-05 18:05:09.918 16188 31026 D WebSocketRepository: Unsubscribing from subscribe_events with data {event_type=entity_registry_updated}
03-05 18:05:09.919 16188 31026 D WebSocketRepository: Sending message 900: {type=unsubscribe_events, subscription=891, id=900}
03-05 18:05:09.920 16188  6449 D WebSocketRepository: Unsubscribing from subscribe_events with data {event_type=device_registry_updated}
03-05 18:05:09.920 16188 31026 D WebSocketRepository: Message number 900 sent
03-05 18:05:09.927 16188  6449 D WebSocketRepository: Sending message 901: {type=unsubscribe_events, subscription=892, id=901}
03-05 18:05:09.929 16188  6449 D WebSocketRepository: Message number 901 sent
03-05 18:05:10.080 16188 30184 D WebSocketRepository: Websocket: onMessage (text)
03-05 18:05:10.102 16188 30184 D WebSocketRepository: Message number 894 received
03-05 18:05:13.348 16188 30184 D WebSocketRepository: Websocket: onMessage (text)
03-05 18:05:13.483 16188 16202 I mpanion.androi: Background concurrent copying GC freed 368544(12MB) AllocSpace objects, 14(7284KB) LOS objects, 49% free, 22MB/45MB, paused 217us total 153.800ms
03-05 18:05:13.488 16188 30184 D WebSocketRepository: Message number 895 received
03-05 18:05:13.488 16188 30184 D WebSocketRepository: Message number 896 received
03-05 18:05:13.488 16188 30184 D WebSocketRepository: Message number 897 received
03-05 18:05:13.489 16188 30184 D WebSocketRepository: Websocket: onMessage (text)
03-05 18:05:13.489 16188 30184 D WebSocketRepository: Message number 898 received
03-05 18:05:13.489 16188 30184 D WebSocketRepository: Message number 899 received
03-05 18:05:13.489 16188 30184 D WebSocketRepository: Message number 900 received
03-05 18:05:13.489 16188 30184 D WebSocketRepository: Message number 901 received
03-05 18:05:20.037 16188  6457 E WebSocketRepository: Websocket: onFailure
03-05 18:05:20.037 16188  6457 E WebSocketRepository: java.net.SocketTimeoutException: failed to connect to home-assistant.iot.domain/10.202.0.13 (port 8123) from /100.64.0.1 (port 48292) after 10000ms
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at libcore.io.IoBridge.connectErrno(IoBridge.java:190)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at libcore.io.IoBridge.connect(IoBridge.java:134)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at java.net.PlainSocketImpl.socketConnect(PlainSocketImpl.java:142)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:390)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:230)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:212)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:436)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at java.net.Socket.connect(Socket.java:621)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.platform.Platform.connectSocket(Platform.kt:128)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:295)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
03-05 18:05:20.037 16188  6457 E WebSocketRepository: 	at java.lang.Thread.run(Thread.java:923)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: Unable to authenticate
03-05 18:05:20.039 16188  9648 E WebSocketRepository: java.net.SocketTimeoutException: failed to connect to home-assistant.iot.domain/10.202.0.13 (port 8123) from /100.64.0.1 (port 48292) after 10000ms
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at libcore.io.IoBridge.connectErrno(IoBridge.java:190)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at libcore.io.IoBridge.connect(IoBridge.java:134)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at java.net.PlainSocketImpl.socketConnect(PlainSocketImpl.java:142)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:390)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:230)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:212)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:436)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at java.net.Socket.connect(Socket.java:621)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.platform.Platform.connectSocket(Platform.kt:128)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:295)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
03-05 18:05:20.039 16188  9648 E WebSocketRepository: 	at java.lang.Thread.run(Thread.java:923)
03-05 18:05:20.040 16188  9648 W WebSocketRepository: Unable to send message, not connected: WebSocketRequest(message={type=config/area_registry/list}, timeout=30000, eventFlow=null, eventTimeout=0, onEvent=null, onResponse=null)
03-05 18:05:20.040 16188  9648 W WebSocketRepository: Unable to send message, not connected: WebSocketRequest(message={type=config/device_registry/list}, timeout=30000, eventFlow=null, eventTimeout=0, onEvent=null, onResponse=null)
03-05 18:05:20.042 16188  6456 W WebSocketRepository: Unable to send message, not connected: WebSocketRequest(message={type=config/entity_registry/list}, timeout=30000, eventFlow=null, eventTimeout=0, onEvent=null, onResponse=null)
03-05 18:05:20.042 16188 31029 W WebSocketRepository: Unable to send message, not connected: WebSocketRequest(message={type=get_states}, timeout=30000, eventFlow=null, eventTimeout=0, onEvent=null, onResponse=null)
03-05 18:05:20.277 16188  9648 W ControlsProviderService: onNext(): control is not stateless. Use the Control.StatelessBuilder() to build the control.
03-05 18:05:20.312 16188  9648 I chatty  : uid=10215(io.homeassistant.companion.android) DefaultDispatch identical 150 lines
03-05 18:05:20.312 16188  9648 W ControlsProviderService: onNext(): control is not stateless. Use the Control.StatelessBuilder() to build the control.
03-05 18:05:29.222 16188  9648 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:05:29.224 16188  9648 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:05:29.227 16188  9648 I chatty  : uid=10215(io.homeassistant.companion.android) DefaultDispatch identical 1 line
03-05 18:05:29.228 16188  9648 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:05:29.229 16188  6456 D WebSocketRepository: Sending message 902: {type=ping, id=902}
03-05 18:05:29.230 16188  6456 D WebSocketRepository: Message number 902 sent
03-05 18:05:29.230 16188  9648 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:05:29.329 16188 30184 D WebSocketRepository: Websocket: onMessage (text)
03-05 18:05:29.330 16188 30184 D WebSocketRepository: Message number 902 received
03-05 18:05:29.348 16188 16188 D HaConProService: publisherFor [1.climate.iot_house_thermostat, 1.climate.master_bathroom, 1.fan.master_bedroom_fan, 1.light.zigbee_group_master_lamp_bulbs, light.family_room_lights, 1.light.piggie_light, 1.light.outside_lights, 1.light.plug_c]
03-05 18:05:29.387 16188  3121 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:05:29.388 16188  3121 D HaConProService: request 100000
03-05 18:05:29.390 16188  9018 D WebSocketRepository: Sending message 903: {type=config/area_registry/list, id=903}
03-05 18:05:29.390 16188  9018 D WebSocketRepository: Message number 903 sent
03-05 18:05:29.391 16188 31029 D WebSocketRepository: Sending message 904: {type=config/device_registry/list, id=904}
03-05 18:05:29.391 16188 31029 D WebSocketRepository: Message number 904 sent
03-05 18:05:29.393 16188 31029 D WebSocketRepository: Sending message 905: {type=config/entity_registry/list, id=905}
03-05 18:05:29.393 16188 31029 D WebSocketRepository: Message number 905 sent
03-05 18:05:29.394 16188 31026 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:05:29.394 16188 31026 D ServerConnectionInfo: Using external URL
03-05 18:05:29.466 16188 30184 D WebSocketRepository: Websocket: onMessage (text)
03-05 18:05:29.467 16188 30184 D WebSocketRepository: Message number 903 received
03-05 18:05:29.628 16188 16188 D IntegrationRepository: isAppLocked(): false. (LockEnabled: false, appActive: false, expireMillis: 1709679910668, currentMillis: 1709679929628)
03-05 18:05:31.022 16188  3121 D HaConProService: cancel
03-05 18:05:31.056 16188 16188 D IntegrationRepository: isAppLocked(): false. (LockEnabled: false, appActive: false, expireMillis: 1709679910668, currentMillis: 1709679931056)
03-05 18:05:31.056 16188 16188 D IntegrationRepository: setAppActive(): true
03-05 18:05:31.088 16188 30184 D WebSocketRepository: Websocket: onMessage (text)
03-05 18:05:31.165 16188 30184 D WebSocketRepository: Message number 904 received
03-05 18:05:31.166 16188 30184 D WebSocketRepository: Message number 905 received
03-05 18:05:31.216 16188  6461 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: false
03-05 18:05:45.131 16188  9018 D LogcatReader: Read logcat for pid 16188
```

--- a/app/src/main/java/io/homeassistant/companion/android/controls/HaControlsProviderService.kt
+++ b/app/src/main/java/io/homeassistant/companion/android/controls/HaControlsProviderService.kt
@@ -94,11 +94,11 @@ class HaControlsProviderService : ControlsProviderService() {
                     return@launch
                 }
 
-                try {
-                    val entities = mutableMapOf<Int, List<Entity<Any>>?>()
-                    val areaForEntity = mutableMapOf<Int, Map<String, AreaRegistryResponse?>>()
-                    serverManager.defaultServers.map { server ->
-                        async {
+                val entities = mutableMapOf<Int, List<Entity<Any>>?>()
+                val areaForEntity = mutableMapOf<Int, Map<String, AreaRegistryResponse?>>()
+                serverManager.defaultServers.map { server ->
+                    async {
+                        try {
                             val getAreaRegistry = async { serverManager.webSocketRepository(server.id).getAreaRegistry() }
                             val getDeviceRegistry = async { serverManager.webSocketRepository(server.id).getDeviceRegistry() }
                             val getEntityRegistry = async { serverManager.webSocketRepository(server.id).getEntityRegistry() }
@@ -119,9 +119,13 @@ class HaControlsProviderService : ControlsProviderService() {
                             }
                             entities[server.id] = entities[server.id].orEmpty()
                                 .sortedWith(compareBy(nullsLast()) { areaForEntity[server.id]?.get(it.entityId)?.name })
+                        } catch (e: Exception) {
+                            Log.e(TAG, "Unable to load entities/registries for server ${server.id} (${server.friendlyName}), skipping", e)
                         }
-                    }.awaitAll()
+                    }
+                }.awaitAll()
 
+                try {
                     val allEntities = mutableListOf<Pair<Int, Entity<Any>>>()
                     entities.forEach { serverEntities ->
                         serverEntities.value?.forEach { allEntities += Pair(serverEntities.key, it) }
@@ -158,7 +162,7 @@ class HaControlsProviderService : ControlsProviderService() {
                             subscriber.onNext(it)
                         }
                 } catch (e: Exception) {
-                    Log.e(TAG, "Error getting list of entities", e)
+                    Log.e(TAG, "Error building list of entities", e)
                 }
                 subscriber.onComplete()
             }

